libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Skills/AdaptiveSkillDialog.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using AdaptiveExpressions.Properties;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Skills
+{
+    public class AdaptiveSkillDialog : SkillDialog
+    {
+        [JsonProperty("$kind")]
+        public const string DeclarativeType = "Microsoft.AdaptiveSkillDialog";
+
+        [JsonConstructor]
+        public AdaptiveSkillDialog([CallerFilePath] string callerPath = "", [CallerLineNumber] int callerLine = 0)
+            : base(new SkillDialogOptions())
+        {
+            DialogOptions.Skill = new BotFrameworkSkill();
+            RegisterSourceLocation(callerPath, callerLine);
+        }
+
+        /// <summary>
+        /// Gets or sets an optional expression which if is true will disable this action.
+        /// </summary>
+        /// <example>
+        /// "user.age > 18".
+        /// </example>
+        /// <value>
+        /// A boolean expression. 
+        /// </value>
+        [JsonProperty("disabled")]
+        public BoolExpression Disabled { get; set; }
+
+        [JsonProperty("botId")]
+        public StringExpression BotId { get; set; } = "=settings.MicrosoftAppId";
+
+        [JsonProperty("skillHostEndpoint")]
```

Comment:
Hi, Gabo, do we plan to support to load skill manifest to begin a skill?

Reply:
Not at the moment, unless I am missing something, what would you use the manifest for at this point?

Reply:
Some of the info can be extracted directly from the manifest file, I am just thinking that will it be more convenient for user to just set the manifest file? But anyway, this also looks good to me.

Reply:
OK, thanks, I will resolve this comment for now. We can review once we have a better idea on how composer will consume and store manifest info.